modules/Walking_module/src/WalkingQPInverseKinematics.cpp

```diff
@@ -232,26 +455,255 @@ bool WalkingQPIK::setDesiredJointPosition(const iDynTree::VectorDynSize& regular
+
+    leftFootCorrection.block(3,0,3,1) = m_kAttFoot * (iDynTree::unskew(iDynTree::toEigen(errorLeftAttitude)));
+
+    Eigen::VectorXd rightFootCorrection(6);
+    rightFootCorrection.block(0,0,3,1) = m_kPosFoot * iDynTree::toEigen((m_rightFootToWorldTransform.getPosition() -
+                                                                         m_desiredRightFootToWorldTransform.getPosition()));
+
+    iDynTree::Matrix3x3 errorRightAttitude = iDynTreeHelper::Rotation::skewSymmetric(m_rightFootToWorldTransform.getRotation() *
+                                                                                     m_desiredRightFootToWorldTransform.getRotation().inverse());
+
+    rightFootCorrection.block(3,0,3,1) = m_kAttFoot * (iDynTree::unskew(iDynTree::toEigen(errorRightAttitude)));
+
+    if((m_desiredLeftFootTwist(0) == m_desiredLeftFootTwist(1)) && (m_desiredLeftFootTwist(0) == 0))
+    {
+        lowerBound.block(0, 0, 6, 1) = iDynTree::toEigen(m_desiredLeftFootTwist);
+        upperBound.block(0, 0, 6, 1) = iDynTree::toEigen(m_desiredLeftFootTwist);
+    }
+    else
+    {
+        lowerBound.block(0, 0, 6, 1) = iDynTree::toEigen(m_desiredLeftFootTwist) - leftFootCorrection;
+        upperBound.block(0, 0, 6, 1) = lowerBound.block(0, 0, 6, 1);
+    }
+
+    if((m_desiredRightFootTwist(0) == m_desiredRightFootTwist(1)) && (m_desiredRightFootTwist(0) == 0))
+    {
+        lowerBound.block(6, 0, 6, 1) = iDynTree::toEigen(m_desiredRightFootTwist);
+        upperBound.block(6, 0, 6, 1) = iDynTree::toEigen(m_desiredRightFootTwist);
+    }
+    else
+    {
+        lowerBound.block(6, 0, 6, 1) = iDynTree::toEigen(m_desiredRightFootTwist) - rightFootCorrection;
+        upperBound.block(6, 0, 6, 1) = lowerBound.block(6, 0, 6, 1);
+    }
+    if(m_useCoMAsConstraint)
+    {
+        lowerBound.block(12, 0, 3, 1) = iDynTree::toEigen(m_desiredComVelocity)
+            - m_kCom * (iDynTree::toEigen(m_comPosition) -  iDynTree::toEigen(m_desiredComPosition));
+        upperBound.block(12, 0, 3, 1) = lowerBound.block(12, 0, 3, 1);
+    }
+
+    setJointVelocitiesBounds();
+}
+
+const iDynTree::VectorDynSize& WalkingQPIK::getSolution() const
+{
+    return m_solution;
+}
+
+const iDynTree::VectorDynSize& WalkingQPIK::getDesiredJointVelocities() const
+{
+    return m_desiredJointVelocitiesOutput;
+}
+
+iDynTree::VectorDynSize WalkingQPIK::getLeftFootError()
+{
+    iDynTree::VectorDynSize output(6);
+    iDynTree::toEigen(output) = iDynTree::toEigen(m_lowerBound).block(0, 0, 6, 1)
+        - iDynTree::toEigen(m_leftFootJacobian) * iDynTree::toEigen(m_solution);
+    return output;
+}
+
+iDynTree::VectorDynSize WalkingQPIK::getRightFootError()
+{
+    iDynTree::VectorDynSize output(6);
+    iDynTree::toEigen(output) = iDynTree::toEigen(m_lowerBound).block(6, 0, 6, 1)
+        - iDynTree::toEigen(m_rightFootJacobian) * iDynTree::toEigen(m_solution);
```

Comment:
Same for the above two methods: fixed size vectors and ``segment``

Reply:
Actually, `getLeftFootError` and `getRightFootError` were implemented just for debugging. I think that we can remove the methods